python/tvm/target/target.py

```diff
@@ -364,9 +394,11 @@ def create(target_str):
     ----
     See the note on :py:mod:`tvm.target` on target string format.
     """
-    if isinstance(target_str, Target):
+    if isinstance(target, Target):
         return target_str
-    if not isinstance(target_str, str):
-        raise ValueError("target_str has to be string type")
+    if isinstance(target, dict):
+        return _ffi_api.TargetFromConfig(target)
+    if isinstance(target, str):
+        return _ffi_api.TargetFromString(target)
```

Comment:
Shall we also do some guessing work as well for strings? If it is a string but seems like a JSON, we can use `json.loads` instead.

```suggestion
    if isinstance(target, str):
        if target.startswith("{"):
          return _ffi_api.TargetFromConfig(json.loads(target))
        else:
          return _ffi_api.TargetFromString(target)
```

Reply:
I would prefer to support the following use case as well:

```python
target = target.create('./target.json') # Input is a string file path.
```
This case could be supported by using `os` to check if the string is a file path or not.

In addition, if we would like to support the JSON string, I personally don't think checking the first character is a proper approach, as you may have spaces or something else. A better way should be just trying to parse the string with JSON parser first and use string parser instead if the JSON parser throws an exception.

Reply:
Yes, I think we should try to find a unified approach to create target from
1) file
2) json string
3) json-like dict
4) legacy raw string
5) tag

Reply:
Given that it is one line to open and read the file. I think we can just allow it takes in string. Notebly there is also a confliction with tag

```target = target.create(open("target.json").read())``` is not as bad

Reply:
I've added a try except block to see if the input string can be parsed to json. We now can accept traditional target string, python dicts, or json strings. Let me know we need other options like tag (not sure what this means).